src/base64Utils.js

```diff
@@ -4,33 +4,38 @@
  * SPDX-License-Identifier: MIT
  */
 
+#if SINGLE_FILE == 2 && !WASM_ASYNC_COMPILATION
+#error SINGLE_FILE == 2 does not work with synchronous WASM compilation (because DecompressionStream is async by nature)
+#endif
+
 #if POLYFILL && (ENVIRONMENT_MAY_BE_SHELL || (ENVIRONMENT_MAY_BE_NODE && MIN_NODE_VERSION < 160000))
 #include "polyfill/atob.js"
 #endif
 
 // Converts a string of base64 into a byte array (Uint8Array).
 function intArrayFromBase64(s) {
-#if ENVIRONMENT_MAY_BE_NODE
+#if ENVIRONMENT_MAY_BE_NODE && SINGLE_FILE != 2
```

Comment:
I think a new internal setting (once that users don't have access too and therefore we are not commited to to supporting) would be good here.  .e.g Something like `USE_GZIP_BLOB_URL_COMPRESSION` added to `settings_internal.js`